startup/90-functions.py

```diff
@@ -1,57 +1,191 @@
+print(f'Loading {__file__}')
 
-def configure_area_det(det,acq_time,exposure,num_exposure=1):
+
+def configure_area_det(det,acq_time,acq_period=None,exposure=None,num_exposures=1):
+    
+    if det.name == 'prosilica':
+        acq_time = min(acq_time,25)
     
     if det.cam.acquire.get() == 0:
         yield from bps.abs_set(det.cam.acquire, 1, wait=True)
 
-    yield from bps.abs_set(det.cam.acquire_time, acq_time, wait=True)
-    acq_time = det.cam.acquire_time.get()
-
-    num_frame = np.ceil(exposure / acq_time)
+    if det.name == 'dexela': 
+        yield from bps.abs_set(det.cam.acquire_time, max(acq_time,0.1), wait=True)
+        acq_time_rbv = det.cam.acquire_time.get()  
+    else:
+        yield from bps.abs_set(det.cam.acquire_time, acq_time, wait=True)
+        acq_time_rbv = det.cam.acquire_time.get()        
+        
+    if det.name == 'dexela': 
+        yield from bps.abs_set(det.cam.acquire_period, acq_time_rbv+0.005, wait=True)
+        acq_period_rbv = det.cam.acquire_period.get()
+    else:
+        if acq_period is None:
+            
+            if det.name == 'blackfly':
+                yield from bps.abs_set(det.cam.acquire_period, 0.1, wait=False)
+            else:
+                yield from bps.abs_set(det.cam.acquire_period, acq_time_rbv, wait=True)
+            acq_period_rbv = det.cam.acquire_period.get()
+        else:
+            if det.name == 'blackfly':
+                yield from bps.abs_set(det.cam.acquire_period, min(1,acq_period), wait=False)
+            else:
+                yield from bps.abs_set(det.cam.acquire_period, acq_period, wait=True)
+            acq_period_rbv = det.cam.acquire_period.get()
```

Comment:
This line and a couple of occurrences above should better use `yield from bps.read()` and then you could extract the needed value from the returned dictionary.

Reply:
I am not sure how to do it. Can you do that for me. Thanks

Reply:
Let's address it via a separate PR. I would like to merge it to this repo's `main` branch shortly.